### PR TITLE
Fix error with external functions and sets.

### LINKED ISSUE #555: External set bridging broken on v0.6.2
I think #544 broke this example of external set bridging. It works with MOI v0.6.1

https://github.com/atoptima/RCSP.jl/blob/master/src/simplebridge.jl

I think it's a scoping issue, but I am not sure about the exact reason.

Here is the error I get:

```
ERROR: LoadError: LoadError: LoadError: UndefVarError: SGT not defined
Stacktrace:
 [1] top-level scope at /Users/itahiri/.julia/packages/MathOptInterface/QO3Mk/src/Utilities/model.jl:636
 [2] top-level scope at /Users/itahiri/.julia/packages/MathOptInterface/QO3Mk/src/Bridges/singlebridgeoptimizer.jl:45
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1038
 [5] include at ./sysimg.jl:29 [inlined]
 [6] include(::String) at /Users/itahiri/.julia/dev/RCSP/src/RCSP.jl:1
 [7] top-level scope at none:0
 [8] include at ./boot.jl:317 [inlined]
 [9] include_relative(::Module, ::String) at ./loading.jl:1038
 [10] include(::Module, ::String) at ./sysimg.jl:29
 [11] include(::String) at ./client.jl:388
 [12] top-level scope at none:0
 [13] include at ./boot.jl:317 [inlined]
 [14] include_relative(::Module, ::String) at ./loading.jl:1038
 [15] include(::Module, ::String) at ./sysimg.jl:29
 [16] include(::String) at ./client.jl:388
 [17] top-level scope at none:0
in expression starting at /Users/itahiri/.julia/dev/RCSP/src/simplebridge.jl:68
in expression starting at /Users/itahiri/.julia/dev/RCSP/src/RCSP.jl:21
```

### PR DESCRIPTION
**Pull request by odow**
Closes #555
Closes #557 

- [x] still need to test bridges

### CODE REVIEW

File: test/Utilities/model.jl [hunk 1]
```diff
@@ -1,5 +1,37 @@
 # TODO: It's hard to find where Model is defined!
 
+# We need to test this in a module at the top level because it can't be defined
+# in a testset. If it runs without error, then we're okay.
+module TestExternal
+    using Base: copy
```
Comment (blegat) [inline test/Utilities/model.jl]: this line shouldn't be needed

File: test/bridge.jl [hunk 2]
```diff
@@ -397,3 +397,57 @@ end
...
     end
 end
+
+# We need to test this in a module at the top level because it can't be defined
+# in a testset. If it runs without error, then we're okay.
+module TestExternalBridge
+    using MathOptInterface
+
+    struct StrictlyGreaterThan <: MathOptInterface.AbstractScalarSet end
+    struct StrictlyGreaterBridge{T} <: MathOptInterface.Bridges.AbstractBridge end
```
Comment (odow) [inline test/bridge.jl]: @blegat do bridges have to have a type parameter `T`? This cause me no end of problems. The only error you get is the unhelpful `Unionall` one.

File: test/bridge.jl [hunk 3]
```diff
@@ -397,3 +397,57 @@ end
...
+        ()
+    )
+end
+
+@testset "@bridge with external components" begin
+    model = SimpleModel{Float64}();
+    @test MOI.supports_constraint(model, MOI.SingleVariable, MOI.GreaterThan{Float64})
+    @test !MOI.supports_constraint(model, MOI.SingleVariable, TestExternalBridge.StrictlyGreaterThan)
+
+    bridge = TestExternalBridge.StrictlyGreater{Float64}(model);
```
Comment (odow) [inline test/bridge.jl]: Follow-up from the above comment. Is it related to the fact that this defines a model using `@model`?

File: test/bridge.jl (hunk 2 above)
Comment (blegat) [reply test/bridge.jl]: Yes, it has to, see https://github.com/JuliaOpt/MathOptInterface.jl/blob/b0477a680fc7ce274af60b0004fc4653cc03bee3/src/Bridges/singlebridgeoptimizer.jl#L46

File: test/bridge.jl (hunk 3 above)
Comment (blegat) [reply test/bridge.jl]: Yes, this is the same type as the MOIU `@model`. This design may be questionable but with light bridges there won't be an internal model anymore

**Pull request merged**